esmvalcore/preprocessor/_time.py

```diff
@@ -711,3 +754,73 @@ def timeseries_filter(cube, window, span,
                                weights=wgts)
 
     return cube
+
+
+def resample_hours(cube, hours):
+    """
+    Converts x-hourly data to y-hourly by eliminating extra timesteps.
+
+    Converts x-hourly data to y-hourly (y > x) by eliminating the extra
+    timesteps. This is intended to be used only with instantaneous values.
+
+    Parameters
+    ----------
+    cube: iris.cube.Cube
+        Input cube.
+    hours: int
+        The frequency of the desired data.
+
+    Returns
+    -------
+    iris.cube.Cube
+        Cube with the new frequency.
+
+    Raises
+    ------
+    ValueError:
+        The specified freuqncy is not a divisor of 24.
+    """
+    if 24 % hours != 0:
+        raise ValueError('The number of hours must be a divisor of 24')
+    hours = range(0, 24, hours)
+    select_hours = iris.Constraint(
+        time=lambda cell: cell.point.hour in hours)
+    return cube.extract(select_hours)
+
+
+def resample_time(cube, month=None, day=None, hour=None):
+    """
+    Change frequency of data by resampling it.
+
+    Converts data from one frequency to another by extracting the timesteps
+    that match the provided month, day and/or hour. This is meant to be used
+    with instantaneous values when computing statistics is not desired.
+
+    Parameters
+    ----------
+    cube: iris.cube.Cube
+        Input cube.
+    month: int, optional
+        Month to extract
+    day: int, optional
+        Day to extract
+    hour: int, optional
+        Hour to extract
+
+    Returns
+    -------
+    iris.cube.Cube
+        Cube with the new frequency.
+    """
+    def compare(date):
+        if month is not None and month != date.month:
+            return False
+        if day is not None and day != date.day:
+            return False
+        if hour is not None and hour != date.hour:
+            return False
+        return True
+
+    resample_constraint = iris.Constraint(
+        time=lambda cell: compare(cell.point))
+    return cube.extract(resample_constraint)
```

Comment:
Wouldn't this be a bit simpler?

```suggestion
    def compare(cell):
        date = cell.point
        if month is not None and month != date.month:
            return False
        if day is not None and day != date.day:
            return False
        if hour is not None and hour != date.hour:
            return False
        return True

    resample_constraint = iris.Constraint(
        time=compare)
    return cube.extract(resample_constraint)
```